app/controllers/posts_controller.rb

```diff
@@ -1,6 +1,6 @@
 class PostsController < ApplicationController
   def index
-    @user = User.includes(:posts, :comments).find(params[:id])
+    @user = User.includes(:posts, :comments).find(params[:user_id])
     @user_posts = @user.posts.includes(:author, :comments, :likes).all
     @comments = Comment.all
```

Comment:
- [ ] Kindly make sure to authorize all actions in the posts controller. You can use ``load_and_authorize_resource`` to do so. You can read more about it [here](https://github.com/CanCanCommunity/cancancan#:~:text=can%20be%20read.-,Controller%20helpers,-The%20authorize!).